app/services/preserver.rb

```diff
@@ -98,13 +81,13 @@ def preserve_children
     end
 
     def preserve_metadata
-      local_md5_checksum = local_checksum_for(file: temp_metadata_file.io)
+      local_checksum = metadata_node.checksum.first
 
       uploaded_file = storage_adapter.upload(
         file: temp_metadata_file.io,
         original_filename: metadata_node.label.first,
         resource: resource,
-        md5: local_md5_checksum
+        md5: local_checksum.md5
```

Comment:
You'll have to convert this to binary. I think it's `Base64.strict_encode64([local_checksum.md5].pack('H*'))`

Reply:
No, I did not test against GCS, I can do this now.

Reply:
I can confirm that `Base64.strict_encode64([local_checksum.md5].pack('H*'))` worked with a personal Google Cloud Storage Bucket.